### ISSUE TITLE
Openstack Builder crashes when using ports

### ISSUE THREAD

**Issue by pmontanari** [hashicorp/packer#6993]
FOR BUGS:

Using packer 1.3.2, when creating an openstack builder using ports, I have the following crash:

Describe the problem and include the following information:
`==> Some builds didn't complete successfully and had errors:
--> openstack: unexpected EOF

==> Builds finished but no artifacts were created.
2018/11/13 04:52:33 packer: panic: runtime error: index out of range
2018/11/13 04:52:33 packer: 
2018/11/13 04:52:33 packer: goroutine 193 [running]:
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/builder/openstack.(*StepRunSourceServer).Run(0xc0002f1b80, 0x28b7f60, 0xc000340340, 0x28b5ca0, 0xc0002f6240, 0x0)
panic: runtime error: index out of range
2018/11/13 04:52:33 packer: 
2018/11/13 04:52:33 packer: goroutine 193 [running]:
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/builder/openstack.(*StepRunSourceServer).Run(0xc0002f1b80, 0x28b7f60, 0xc000340340, 0x28b5ca0, 0xc0002f6240, 0x0)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/builder/openstack/step_run_source_server.go:50 +0x10d5
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc000340280, 0x28b5ca0, 0xc0002f6240)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/helper/multistep/basic_runner.go:72 +0x222
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/builder/openstack.(*Builder).Run(0xc0000d3100, 0x28bdbc0, 0xc00000a440, 0x28ab1e0, 0xc0001ce2d0, 0x28b8320, 0xc0001d0058, 0x4bcdac, 0x1e89c80, 0x1e89c80, ...)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/builder/openstack/builder.go:151 +0xbb3
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0001cc6e0, 0x1, 0xc0001dc100, 0x0, 0x0)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/packer/rpc/builder.go:94 +0x1cd
2018/11/13 04:52:33 packer: reflect.Value.call(0xc0001d2840, 0xc0001d80b0, 0x13, 0x25d47be, 0x4, 0xc000164f18, 0x3, 0x3, 0xc0001ba0c0, 0x0, ...)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/reflect/value.go:447 +0x449
2018/11/13 04:52:33 packer: reflect.Value.Call(0xc0001d2840, 0xc0001d80b0, 0x13, 0xc000553f18, 0x3, 0x3, 0x0, 0x0, 0x0)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/reflect/value.go:308 +0xa4
2018/11/13 04:52:33 packer: net/rpc.(*service).call(0xc0001ba240, 0xc0001ec190, 0xc00029e240, 0xc00029e250, 0xc000146680, 0xc0001ccb20, 0x1efb7a0, 0xc0001dc0fc, 0x18a, 0x1e89c80, ...)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/net/rpc/server.go:384 +0x14e
2018/11/13 04:52:33 packer: created by net/rpc.(*Server).ServeCodec
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/net/rpc/server.go:481 +0x47e
2018/11/13 04:52:33 ui error: Build 'openstack' errored: unexpected EOF
2018/11/13 04:52:33 Builds completed. Waiting on interrupt barrier...
2018/11/13 04:52:33 machine readable: error-count []string{"1"}
2018/11/13 04:52:33 ui error: 
==> Some builds didn't complete successfully and had errors:
2018/11/13 04:52:33 machine readable: openstack,error []string{"unexpected EOF"}
2018/11/13 04:52:33 ui error: --> openstack: unexpected EOF
2018/11/13 04:52:33 ui: 
==> Builds finished but no artifacts were created.
2018/11/13 04:52:33 waiting for all plugin processes to complete...
2018/11/13 04:52:33 /var/lib/jenkins/jobs/packer_controller_vm_multi/branches/Dev/workspace/jenkins/build/controller_vm/packer: plugin process exited
2018/11/13 04:52:33 /var/lib/jenkins/jobs/packer_controller_vm_multi/branches/Dev/workspace/jenkins/build/controller_vm/packer: plugin process exited

2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/builder/openstack/step_run_source_server.go:50 +0x10d5
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc000340280, 0x28b5ca0, 0xc0002f6240)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/helper/multistep/basic_runner.go:72 +0x222
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/builder/openstack.(*Builder).Run(0xc0000d3100, 0x28bdbc0, 0xc00000a440, 0x28ab1e0, 0xc0001ce2d0, 0x28b8320, 0xc0001d0058, 0x4bcdac, 0x1e89c80, 0x1e89c80, ...)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/builder/openstack/builder.go:151 +0xbb3

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2018/11/13 04:52:33 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0001cc6e0, 0x1, 0xc0001dc100, 0x0, 0x0)
2018/11/13 04:52:33 packer: /Users/azr/go/src/github.com/hashicorp/packer/packer/rpc/builder.go:94 +0x1cd
2018/11/13 04:52:33 packer: reflect.Value.call(0xc0001d2840, 0xc0001d80b0, 0x13, 0x25d47be, 0x4, 0xc000164f18, 0x3, 0x3, 0xc0001ba0c0, 0x0, ...)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/reflect/value.go:447 +0x449
2018/11/13 04:52:33 packer: reflect.Value.Call(0xc0001d2840, 0xc0001d80b0, 0x13, 0xc000553f18, 0x3, 0x3, 0x0, 0x0, 0x0)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/reflect/value.go:308 +0xa4
2018/11/13 04:52:33 packer: net/rpc.(*service).call(0xc0001ba240, 0xc0001ec190, 0xc00029e240, 0xc00029e250, 0xc000146680, 0xc0001ccb20, 0x1efb7a0, 0xc0001dc0fc, 0x18a, 0x1e89c80, ...)
2018/11/13 04:52:33 packer: /usr/local/Cellar/go/1.11.1/libexec/src/net/rpc/server.go:384 +0x14e
2018/11/13 04:52:33 packer: created by net/rpc.(*Server).ServeCodec
Tue Nov 13 04:52:33 CST 2018`

The builder conf, looks like that:
`{
            "type": "openstack",
            "identity_endpoint": "http://xxxxxxxxx:5000/v2.0",
            "tenant_name": "my_tenant",
            "username": "my_user",
            "password": "my_password",
            "region": "RegionOne",
            "ssh_username": "root",
            "ssh_keypair_name": "my_key",
            "image_name": "test",
            "source_image": "85544a3a-0b46-4e2e-a06b-20595e39d1a6",
            "networks": [
                "885a0415-7163-4ec7-b517-fc45639cc6d1"
            ],
            "ports": [
                 "1177d1ba-0aae-4fd0-8c13-4c0849ec2ee7"
            ],
            "user_data_file": "userdata.txt",
            "flavor": "m1.medium",
            "insecure": "true"
        }`

**Comment by xzxzzxzxzca** [comment]
I think I'm seeing the same problem. Specs: FreeBSD 12.0, packer-1.3.3, builder config (stripped):
```
  {
      "type": "amazon-ebs",
      "instance_type": "t2.medium",
      "source_ami": "ami-...",
      "ami_name": "...",
      "ami_virtualization_type": "hvm",
      "iam_instance_profile": "...",

      "region": "eu-central-1",
      "availability_zone": "eu-central-1a",
      "subnet_id": "...",
      "security_group_ids": ["sg-..."],

      "communicator": "ssh",
      "ssh_username": "ec2-user",

      "force_deregister": true,
      "force_delete_snapshot": true,

      "tags": {
      ...
      },
      "run_tags": {
        ...
      }
    }
```

Crash log: [packer-crash.log](https://github.com/hashicorp/packer/files/3266533/packer-crash.log)

The config hasn't been touched for quite a while, neither did we update packer itself, but the error has never been noticed before.

**Comment by rickard-von-essen** [comment]
@xzxzzxzxzca That's a different issue and you are using an old version of packer. Please upgrade and report a new issue if you can reproduce on the latest version.
